lib/rules/html-self-closing-style.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * @author Toru Nagashima
+ * @copyright 2016 Toru Nagashima. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+/**
+ * Kind strings.
+ * This strings wil be displayed in error messages.
+ */
+const KIND = Object.freeze({
+  NORMAL: 'HTML elements',
+  VOID: 'HTML void elements',
+  COMPONENT: 'Vue.js custom components',
+  SVG: 'SVG elements',
+  MATH: 'MathML elements'
+})
+
+/**
+ * Normalize the given options.
+ * @param {Object|undefined} options The raw options object.
+ * @returns {Object} Normalized options.
+ */
+function parseOptions (options) {
+  return {
+    [KIND.NORMAL]: (options && options.html && options.html.normal) || 'never',
```

Comment:
Or sindresorhus' [dot-prop](https://www.npmjs.com/package/dot-prop). However, I'm not sure if this is enough worth adding a new dependency. This may be not stylish, but I think enough readable.